### PR TITLE
Add a nav menu

### PR DESCRIPTION
**Pull request by j-koziel**
It _broke_ ✨

### CODE REVIEW

**Comment by j-koziel** [comment]
For reference this is the error that I am encountering:
image

File: frontend/app/layout.tsx [hunk 1]
```diff
@@ -1,53 +1,33 @@
...
           defaultTheme="dark"
           enableSystem
           disableTransitionOnChange
         >
-          <div className="relative flex min-h-screen flex-col">
-            <div className="flex-1">{children}</div>
-            {/* SITE FOOTER */}
-            <Toaster />
-          </div>
+          <RootNavMenu />
```
Comment (JudeDavis1) [inline frontend/app/layout.tsx:28]: Try to use this design pattern (if in a SSR component, any client-side rendered components need to be components in another file)

File: frontend/components.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "rsc": true,
   "tsx": true,
   "tailwind": {
-    "config": "tailwind.config.js",
+    "config": "tailwind.config.ts",
```
Comment (JudeDavis1) [inline frontend/components.json:7]: Noice 👍

File: frontend/app/login/page.tsx [hunk 3]
```diff
@@ -38,14 +38,14 @@ export default function Login() {
...
               }}
-              className="hover:shadow-md font-bold"
+              className="hover:shadow-lg hover:shadow-white font-bold"
             >
               Login
-            </span>
+            </p>
           </p>
           <RegisterForm />
         </div>
```
Comment (JudeDavis1) [inline frontend/app/login/page.tsx:51]: In Next.js `page.tsx` should be barebones and reserved for server-side rendering. Put the child (client-side) components in the corresponding `components/` folder and put the "use client" directive on them.

**Review by JudeDavis1 — CHANGES_REQUESTED** [review]

File: frontend/app/layout.tsx (hunk 1 above)
Comment (j-koziel) [reply frontend/app/layout.tsx:28]: ahh ok, will do

File: frontend/app/login/page.tsx (hunk 3 above)
Comment (j-koziel) [reply frontend/app/login/page.tsx:51]: say no more

File: frontend/app/root-nav-menu.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    <div className="min-h-[5vh] m-4 flex flex-row justify-end items-center">
+      <NavigationMenu>
+        <NavigationMenuList>
+          <div className="flex flex-row gap-4 items-center justify-center">
+            <NavigationMenuItem>
+              <NavigationMenuTrigger>Features</NavigationMenuTrigger>
+              <NavigationMenuContent>
+                <div className="flex flex-col items-center justify-center w-32 gap-2 p-4">
+                  <NavigationMenuLink className="p-2 rounded transition-[0.5s] hover:bg-slate-700">
+                    Feature 1
```
Comment (j-koziel) [inline frontend/app/root-nav-menu.tsx:25]: what features do you think would be best to put here?

Comment (JudeDavis1) [reply frontend/app/root-nav-menu.tsx:25]: Hmm. I presume all or some of what are in the wiki: https://github.com/j-koziel/cong-manager/wiki/Project-scope

Comment (j-koziel) [reply frontend/app/root-nav-menu.tsx:25]: Ok sounds good

File: frontend/app/login/page.tsx [hunk 5]
```diff
@@ -16,7 +16,7 @@ export default function Login() {
   };
 
   return (
-    <div className="min-h-screen min-w-full flex flex-col justify-center items-center">
+    <div className="min-h-[95vh] min-w-full flex flex-col justify-center items-center">
```
Comment (JudeDavis1) [inline frontend/app/login/page.tsx:19]: Can we remove the hardcoded values? What happens when we leave as `screen`?

**Review by JudeDavis1 — CHANGES_REQUESTED** [review]
Just one comment!

**Pull request merged**